app/Http/Middleware/ThrottleSignup.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Carbon\Carbon;
+use App\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+
+class ThrottleSignup
+{
+    public function handle(Request $request, Closure $next, string $limit, string $range)
+    {
+        if (!$limit || !$range) {
+            return $next($request);
+        }
+
+        $lookback = Carbon::now()->sub(new \DateInterval($range));
+        $recentSignups = User::where('created_at', '>=', $lookback)->count();
+        if ($recentSignups >= intval($limit)) {
+            Log::warning("WARN_SIGNUP_THROTTLED: Given limit of '$limit' in range '$range' was exceeded, attempted account creation was blocked.");
+            return response()->json(
+                ["errors" => "Due to high load, we're currently not able to create an account for you. Please try again tomorrow or reach out through our contact page."],
+                429
```

Comment:
Not sure if a 429 is super correct here, but everything else 4xx fits even worse https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Reply:
Maybe 503 would be a better choice as it's not really the client's fault.

Reply:
Yeah, I'll sneak this in still before merging.